python/eos/pyhf.py

```diff
@@ -0,0 +1,37 @@
+import eos
+import numpy as np
+
+class PyhfLogLikelihood:
+    def __init__(self, cache, workspace, parameter_map={}):
+        self.cache = cache
+        self.parameters = cache.parameters()
+        self.parameter_map = parameter_map
+
+        self.workspace = workspace
+        self.model = workspace.model()
+        self.data = np.array(workspace.data(self.model, include_auxdata=False))
+
+        self.varied_parameters = []
+        for pn, v in zip(self.model.config.par_names, self.model.config.suggested_init()):
+            if pn in parameter_map:
+                qn = eos.QualifiedName(parameter_map[pn])
+            else:
+                qn = eos.QualifiedName('pyhf::' + pn)
+
+            p = self.parameters.declare(qn, v)
```

Comment:
This is not quite OK. We should only declare this parameter if it is not declared yet!
```
if self.parameters.had(qn):
    p = self.parameters[qn]
else:
    p = self.parameters.declare(qn, v)
```

Reply:
I fat-fingered this one as well: ``self.parameters.has``.